### ISSUE TITLE
Game Crash on AI founding settlement

### ISSUE THREAD

**Issue by MasterofJOKers** [unknown-horizons/unknown-horizons#2418]
```
Traceback (most recent call last):
  File "./run_uh.py", line 380, in <module>
    main()
  File "./run_uh.py", line 189, in main
    ret = horizons.main.start(options)
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/main.py", line 281, in start
    horizons.globals.fife.run()
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/engine/engine.py", line 276, in run
    self.loop()
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/engine/engine.py", line 288, in loop
    f()
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/timer.py", line 112, in check_tick
    f(self.tick_next_id)
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/scheduler.py", line 96, in tick
    callback.callback()
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/world/units/ship.py", line 108, in _move_tick
    super(Ship, self)._move_tick(resume)
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/world/units/movingobject.py", line 204, in _move_tick
    self._movement_finished()
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/world/units/ship.py", line 129, in _movement_finished
    super(Ship, self)._movement_finished()
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/world/units/movingobject.py", line 155, in _movement_finished
    self.move_callbacks.execute()
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/util/python/weakmethodlist.py", line 51, in execute
    callback()
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/util/python/weakmethod.py", line 38, in __call__
    return self.function(*args, **kwargs)
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/util/python/callback.py", line 47, in __call__
    return self.callback(*self.args, **self.kwargs)
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/ai/aiplayer/mission/foundsettlement.py", line 92, in _reached_destination_area
    self.warehouse = builder.execute(self.land_manager, ship=self.ship)
  File "/run/Programme/Programme/unknown-horizons/unknown-horizons/horizons/ai/aiplayer/basicbuilder.py", line 79, in execute
    assert build_position.buildable
AssertionError
AL lib: (EE) alc_cleanup: 1 device not closed
```

**Comment by MasterofJOKers** [comment]
I've got a savegame where this happens directly afterwards. This is probably not fixable for already broken savegames. In this savegame, the AI tries to found a settlement on a resource deposit.

**Comment by cpdef** [comment]
I had exactly the same error.
I compiled the game yesterday from source (newest version) on gentoo linux.
I saved the game before the crash, so everytime i load the game at this point, it crash after a few seconds.

here are some additional informations:
for the build_position variable:
`
build_position.action = idle
build_position.buildable = False
build_position.position = Rect(o:(42,168),w:3,h:3)
build_position.problem = (5, u'Dieser Bereich wird bereits von einem anderen Geb\xe4ude besetzt.')
build_position.rotation = 315
build_position.tearset = []
`

**Comment by MasterofJOKers** [comment]
Did you look at the location on you map? Why can't the AI build a warehouse at that location? Is it a resource deposit, too? (You can press 'H' to show the coordinates on the map.)

**Comment by LinuxDonald** [comment]
Test it with master please maybe i got fixed that: https://github.com/unknown-horizons/unknown-horizons/commit/6a8c988d32cb85dcc2146ea638b14f5b9587ccb1

**Comment by cpdef** [comment]
Ok thx, i have copyed in the relevant sections.
Now the game has no error anymore.
At this place was a 'stonedeposit'.

Must recompile the game, if i wanna add all new changes?
Or what do u do? Only copy some folder?

**Comment by LinuxDonald** [comment]
Make:"git pull" when you use an git checkout

**Comment by MasterofJOKers** [comment]
My savegame still crashes with the latest master.

**Comment by MasterofJOKers** [comment]
But maybe @cpdef has a savegame, where the AI hasn't decided where to place the warehouse, yet. Then it might be fixed. In my savegame, the AI's ship is already moving, so it will always load the wrong planned location from the savegame.

**Comment by cpdef** [comment]
One game crashes now, the second crashed after 10 Minutes or so, i thinked at first it works.
:( sorry for telling not truth.
So i have now the same Problem as before.
But I'am not home next weeks, so i can not help to fix a bug.
But i can upload my games in a repository and add a link here, so other people can test.

**Comment by LinuxDonald** [comment]
Have you tested it with old savegames or new game?

**Comment by cpdef** [comment]
Old. So should i retest it, with a new game?
I start a new game and make like a longtime test ok.

**Comment by LinuxDonald** [comment]
That would be good :-)

**Comment by MasterofJOKers** [comment]
Old or new really shouldn't matter, as long as the AI hasn't already sent a ship to the wrong destination. Using a previous step of my savegame, the AI decides differently all the time, so sometimes it crashes as in the later savegame and sometimes it doesnt.

From digging through the code yesterday, I don't think @LinuxDonald 's commit is really related, as it doesn't touch any paths I've followed on my way. I'm still trying to debug it and my best guess is, that there's a `check_build()` call missing in `horizons.ai.aiplayer.mission.foundsettlement.FoundSettlement.find_warehouse_location()`. But I haven't looked at the specific `check_build()`, yet, to see if a call would fix the problem.

**Comment by MasterofJOKers** [comment]
My assumption would be, that there hasn't been a ship sent, if the crash happens after 10 minutes. 

@cpdef: It's not your fault, it's the randomness of decision-making, that made you assume everything worked.

**Comment by MasterofJOKers** [comment]
I'll close this, as it seems fixed after writing a test. But it is possible, that old savegames with an already sent ship will still try to found a settlement in the wrong place.

**Comment by MasterofJOKers** [comment]
Just found out, that it was still possible for the AI to choose a wrong location. Should be fixed with b670f52.
